File: 程式設計：資料結構答案檔/C#/401.cs
using System;
using System.IO;
using System.Linq;

class GraphReader
{
    static void Main(string[] args)
    {
        string filename = "read.txt";
        int[][] graph;
        int numVertices = 0;
        
        try
        {
            var lines = File.ReadAllLines(filename);
            numVertices = lines[0].Split(',').Length;
            graph = new int[numVertices][];
            
            for (int i = 0; i < lines.Length; i++)
            {
                graph[i] = lines[i].Split(',')
                                   .Select(int.Parse)
                                   .ToArray();
            }

            int[] degrees = new int[numVertices];
            int maxWeight = 0, minWeight = int.MaxValue;

            for (int i = 0; i < numVertices; i++)
            {
                int weightSum = 0;
                for (int j = 0; j < numVertices; j++)
                {
                    if (graph[i][j] > 0)
                    {
                        degrees[i]++;
                        weightSum += graph[i][j];
                    }
                }
                maxWeight = Math.Max(maxWeight, weightSum);
                minWeight = Math.Min(minWeight, weightSum);
            }

            Console.Write("Degree:");
            for (int i = 0; i < numVertices - 1; i++)
            {
                Console.Write($"{degrees[i]},");
            }
            Console.WriteLine(degrees[numVertices - 1]);
            Console.WriteLine(maxWeight - minWeight);
        }
        catch (Exception e)
        {
            Console.WriteLine($"Error reading file: {e.Message}");
        }
    }
}

File: 程式設計：資料結構答案檔/Python/401.py
with open(r'read.txt', 'r') as file:
    graph = []
    
    for line in file.readlines():
        lst = line.strip().split(',')
        lst = [eval(x) for x in lst]
        graph.append(lst)

degree = []
weight = []

for node in graph:
    deg = sum([1 if x>0 else 0 for x in node])
    degree.append(str(deg))
    weight.append(sum(node))

print(f"Degree:{','.join(degree)}")
print(max(weight) - min(weight))

